staging/src/k8s.io/code-generator/cmd/api-linter/linters/optionalfield.go

```diff
@@ -0,0 +1,33 @@
+package linters
+
+import (
+	"k8s.io/gengo/types"
+)
+
+type optionalFieldAPIConvention struct{}
+
+func (c optionalFieldAPIConvention) Name() string {
+	return "optional_fields_linter_pointers"
+}
+
+func (c optionalFieldAPIConvention) Message() string {
+	return "optional fields should be pointers"
+}
+
+func (c optionalFieldAPIConvention) Validate(t *types.Type) ([]string, error) {
+	violationIDs := make([]string, 0)
+
+	// Only validate struct type and ignore the rest
+	switch t.Kind {
+	case types.Struct:
+		for _, m := range t.Members {
+			if hasAPITagValue(m.CommentLines, tagValueFalse) {
+				continue
+			}
+			if hasOptionalTag(&m) && m.Type.Kind != types.Pointer {
```

Comment:
With this implementation, maps and slices will be considered violation which is wrong. Our API convention is different from what we are checking here for maps and slices. The type need to be nillable (map, slice, or pointer to other types I guess).